src/Comment.php

```diff
@@ -204,6 +204,14 @@ public static function parse(string $json, bool $assoc = false, int $depth = 512
 
     public static function parseFromFile(string $file, bool $assoc = false, int $depth = 512, int $options = 0)
     {
+        if (!file_exists($file)) {
```

Comment:
this check should not be necessary. is_file check must be enough

Reply:
I guess that depends on how specific you want to be with exceptions. If we don't split it, the exception must be something like "[given arg] is not a file or does not exist", which is a bit more vague.

But maybe more specific testing needs to be done on the "client" side, if the difference matters. I can see the argument that it's not the responsiblity of this class to check for the difference.

Reply:
Updated all points